projects/app-lob/src/app/grid-finjs/main.component.ts

```diff
@@ -85,10 +95,15 @@ export class FinJSDemoComponent implements OnDestroy {
 
     public setChartData(args: Stock[]): void {
         this.chartData = [];
-        args.forEach(row => {
-            this.chartData.push(row);
-            this.chart.notifyInsertItem(this.chartData, this.chartData.length - 1, row);
+        this.finGrid.grid.data.filter(data => {

```

Comment:
Instead of going through all grid data and compare the id, grid API should be used for such cases. It is more intuitive and more readable.